SYCL/Basic/accessor/accessor.cpp

```diff
@@ -60,6 +60,57 @@ template <typename Acc> struct Wrapper2 {
 
 template <typename Acc> struct Wrapper3 { Wrapper2<Acc> w2; };
 
+template <typename T> void TestAccSizeFuncs(const std::vector<T> &vec) {
+  std::vector<size_t> res(vec.size()), locRes(vec.size());
+  try {
+    sycl::queue q;
+    sycl::buffer<T> buf(vec.data(), vec.size());
+    sycl::buffer<size_t> bufRes(res.data(), res.size());
+    sycl::buffer<size_t> bufLocRes(locRes.data(), locRes.size());
+
+    q.submit([&](sycl::handler &cgh) {
+      auto acc = buf.template get_access<sycl::access::mode::read>(cgh);
+      auto accRes = bufRes.get_access<sycl::access::mode::write>(cgh);
+      auto accLocRes = bufLocRes.get_access<sycl::access::mode::write>(cgh);
+      sycl::local_accessor<T, 1> localAcc(vec.size(), cgh);
+      cgh.single_task([=]() {
+        accRes[0] = acc.byte_size();
```

Comment:
I believe this is relatively hard to maintain. Would something like this (pseudocode) be better:

```
  auto Size = vec.size();
  ...
  sycl::buffer<bool, 1> b(1);
  sycl::accessor acc(b, cgh);  // Better to use SYCL2020 version, I've heard that's what language people see as future.
  cgh.single_task([=]() {
    acc[0] = true;
    acc[0] &= (acc.size() == Size);
  });
```

Or, if we want more granularity than simple pass/fail:

```
  auto Size = vec.size();
  ...
  sycl::buffer<bool, 1> b(1000);
  sycl::accessor acc(b, cgh);  // Better to use SYCL2020 version, I've heard that's what language people see as future.
  cgh.single_task([=]() {
    int Idx = 0;
    acc[Idx++] = (acc.size() == Size);
    ...
   // Maybe store Idx into another acc to know how many elems to verify on host.
  });
```

Reply:
But, using original way we can determine which method is broken. BTW I've reduced & groomed this function a little, could you take one more look?